homie-influx/src/main.rs

```diff
@@ -31,64 +32,86 @@ async fn main() -> Result<(), eyre::Report> {
 
         let influxdb_client = get_influxdb_client(&config.influxdb, &mapping.influxdb_database)?;
 
-        let handle = spawn_homie_poll_loop(event_loop, controller.clone(), influxdb_client);
+        let handle = spawn_homie_poll_loop(
+            event_loop,
+            controller.clone(),
+            influxdb_client,
+            config.mqtt.reconnect_interval,
+        );
         controller.start().await?;
-        join_handles.push(handle.map(|res| Ok(res??)));
+        join_handles.push(handle);
     }
 
-    simplify_unit_vec(try_join_all(join_handles).await)
+    simplify_unit_vec(try_join_all(join_handles).await)?;
```

Comment:
No need to simplify it you are throwing the result away

Reply:
Done.